### PR TITLE
[MOBL-814] enable in-memory cache

### PR DESCRIPTION
**Pull request by rahulrvp**

### CODE REVIEW

File: android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+            if (reqWidth == 0) reqWidth = options.outWidth;
+            if (reqHeight == 0) reqHeight = options.outHeight;
+
+            BlueshiftLogger.d(TAG, "reqWidth: " + reqWidth + ", reqHeight: " + reqHeight);
+
+            options.inSampleSize = calculateInSampleSize(options, reqWidth, reqHeight);
+
+            options.inJustDecodeBounds = false;
+
+            Bitmap raw = BitmapFactory.decodeStream(new URL(url).openStream(), new Rect(), options);
```
Comment (ketanshikharebsft) [inline android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java:264]: I am assuming that the decodeStream is for downloading the image. If that's the case, why are we downloading it twice?

File: android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            bitmap = getFromMemCache(key);
+            if (bitmap == null) {
+                bitmap = getFromDiskCache(context, key);
+                if (bitmap == null) {
+                    bitmap = downloadScaledBitmap(url, reqWidth, reqHeight);
+                    if (bitmap != null) {
+                        addToMemCache(key, bitmap);
+                        addToDiskCache(context, key, bitmap);
+                    } else {
+                        BlueshiftLogger.d(TAG, "Loading image from network");
```
Comment (ketanshikharebsft) [inline android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java:75]: Could you please confirm the placement of the log `loading from the network`?

**Review by ketanshikharebsft — APPROVED** [review]
LGTM

File: android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java (hunk 1 above)
Comment (rahulrvp) [reply android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java:264]: in the first method call, we are only getting the meta info like bounds and other stuff. we don't download the content (ie; bitmap). now, these details will be used after calculating a new sample size into the second call. this helps us in reducing the size of the bitmap without compromising much on the quality. [reference](https://developer.android.com/topic/performance/graphics/load-bitmap#:~:text=Load%20a%20Scaled%20Down%20Version%20into%20Memory,-Now%20that%20the&text=Amount%20of%20memory%20you%20are,density%20of%20the%20current%20device.)

File: android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java (hunk 2 above)
Comment (rahulrvp) [reply android-sdk/src/main/java/com/blueshift/BlueshiftImageCache.java:75]: good find man. it should be outside this else block. I'll change

**Pull request merged**